### PR TITLE
Add Educator AAA to educator node

### PR DESCRIPTION
**Pull request by Lucus16**
- [ ] Handle database migrations
- [ ] Automatically set up aws bucket?
- [ ] Automatically generate `/var/lib/educator_aaa/secret_key_base.env` if it doesn't exist yet
- [ ] Test

### CODE REVIEW

File: modules/services/educator_aaa.nix [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      createHome = true;
+      isSystemUser = true;
+    };
+
+    services.postgresql = {
+      enable = true;
+      package = pkgs.postgresql100;
+      initialScript = pkgs.writeText "educator_aaa.psql" ''
+        CREATE ROLE "educator_aaa" LOGIN PASSWORD 'educator_aaa';
+        CREATE DATABASE "educator_aaa" OWNER 'educator_aaa';
```
Comment (Lucus16) [inline modules/services/educator_aaa.nix]: I think this conflicts with any other services trying to create databases for themselves, but there doesn't seem to be an option for this that doesn't conflict. Please advice.

Comment (mkaito) [reply modules/services/educator_aaa.nix]: Yea, `initialScript` is only ever run once for a postgres deployment. But I think it's type `lines`, so it'll merge things before running.

However, `intialScript` is less than reliable for setting up a known state of the database, and I don't like it at all.

File: modules/services/educator_aaa.nix [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+      script = ''
+        exec ${package}/bin/educator_aaa foreground
+      '';
+
+      serviceConfig = {
+        EnvironmentFile = toString keys.educator_aaa;
+        User = "educator_aaa";
+        WorkingDirectory = "/var/lib/${stateDir}";
+        StateDirectory = stateDir;
```
Comment (mkaito) [inline modules/services/educator_aaa.nix]:
```suggestion
        StateDirectory = stateDirName;
```

File: deployments/cluster/educator.nix [hunk 3]
```diff
@@ -20,6 +20,11 @@ in
     4040        # Educator HTTP API
   ];
 
+  services.educator_aaa = {
+    enable = true;
+    port = 4000;
```
Comment (mkaito) [inline deployments/cluster/educator.nix]:
```suggestion
```

File: modules/services/educator_aaa.nix [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  options.services.educator_aaa = {
+    enable = mkEnableOption "Educator AAA";
+
+    virtualHost = mkOption {
+      default = "educator_aaa";
+      type = types.nullOr types.string;
+    };
+
+    port = mkOption {
+      type = types.int;
```
Comment (mkaito) [inline modules/services/educator_aaa.nix]:
```suggestion
      type = types.int;
      default = 4000;
```

**Review by mkaito — DISMISSED** [review]

**Pull request closed without merging**